src/etc/inc/util.inc

```diff
@@ -1932,24 +1932,63 @@ function mac_format($clientmac) {
 	}
 }
 
-function resolve_retry($hostname, $retries = 5) {
-
-	if (is_ipaddr($hostname)) {
-		return $hostname;
-	}
+function resolve_retry($hostname, $protocol = 'inet', $retries = 5, $numrecords = 1) {
 
+	$recresult = array();
+	$returnres = array();
 	for ($i = 0; $i < $retries; $i++) {
-		// FIXME: gethostbyname does not work for AAAA hostnames, boo, hiss
-		$ip = gethostbyname($hostname);
-
-		if ($ip && $ip != $hostname) {
-			/* success */
-			return $ip;
+		switch ($protocol) {
+			case 'any':
+				$checkproto = 'is_ipaddr';
+				$dnsproto = DNS_ANY;
+				$dnstype = array('A', 'AAAA');
+				break;
+			case 'inet6':
+				$checkproto = 'is_ipaddrv6';
+				$dnsproto = DNS_AAAA;
+				$dnstype = array('AAAA');
+				break;
+			case 'inet': 
+			default:
+				$checkproto = 'is_ipaddrv4';
+				$dnsproto = DNS_A;
+				$dnstype = array('A');
+				break;
+		}
+		if ($checkproto($hostname)) {
+			return $hostname;
+		}
+		$dnsresult = dns_get_record($hostname, $dnsproto);
+		foreach ($dnsresult as $dnsrec => $ip) {
+			if (is_array($ip)) {
+				if (in_array($ip['type'], $dnstype)) {
+				    if ($checkproto($ip['ip'])) { 
+					    $recresult[] .= $ip['ip'];
+				    }
+				    if ($checkproto($ip['ipv6'])) { 
+					    $recresult[] .= $ip['ipv6'];
```

Comment:
Why is this using `.=`? There is nothing to add to, it should just be `=`.

Reply:
fixed